component/src/main/java/io/siddhi/extension/script/js/EvalFunctionExtension.java

```diff
@@ -0,0 +1,164 @@
+                        DataType.STRING, DataType.BOOL, DataType.OBJECT}),
+        examples =
+        @Example(
+                syntax = "script:eval(\"700 > 800\", 'bool')",
+                description = "" +
+                        "In this example, the expression 700 > 800 will be evaluated and " +
+                        "return result as false because user specified return type as bool.")
+)
+public class EvalFunctionExtension extends FunctionExecutor {
+
+    private static final String ENGINE_NAME = "nashorn";
+    private ScriptEngine engine;
+    private Attribute.Type returnType = Attribute.Type.OBJECT;
+
+    @Override
+    protected StateFactory<State> init(ExpressionExecutor[] expressionExecutors,
+                                       ConfigReader configReader,
+                                       SiddhiQueryContext siddhiQueryContext) {
+        int executorsCount = expressionExecutors.length;
+
+        if (executorsCount != 2) {
+            throw new SiddhiAppValidationException("Invalid number of arguments passed to "
+                    + "script:eval() function. Required exactly 2, but found " + executorsCount);
+        }
+        ExpressionExecutor executor1 = expressionExecutors[0];
+        ExpressionExecutor executor2 = expressionExecutors[1];
+
+        if (executor1.getReturnType() != STRING) {
+            throw new SiddhiAppValidationException("Invalid parameter type found for the first argument of "
+                    + "script:eval()() function, required " + STRING.toString() + ", but found "
+                    + executor1.getReturnType().toString());
+
+        }
+        if (!(executor2 instanceof ConstantExpressionExecutor)) {
+            throw new SiddhiAppValidationException("The second argument has to be a string constant specifying " +
+                    "one of the supported data types "
+                    + "(int, long, float, double, string, bool)");
+        } else {
+            String type = attributeExpressionExecutors[1].execute(null).toString();
+            switch (type.toLowerCase(Locale.ENGLISH)) {
+                case "int":
+                    returnType = Attribute.Type.INT;
+                    break;
+                case "long":
+                    returnType = Attribute.Type.LONG;
+                    break;
+                case "float":
+                    returnType = Attribute.Type.FLOAT;
+                    break;
+                case "double":
+                    returnType = Attribute.Type.DOUBLE;
+                    break;
+                case "bool":
+                    returnType = Attribute.Type.BOOL;
+                    break;
+                case "string":
+                    returnType = Attribute.Type.STRING;
+                    break;
+                default:
+                    throw new SiddhiAppValidationException("Type must be one of int, long, float, double, bool, " +
```

Comment:
Rephrase the error message as "Invalid return type found: <type>. Return types supported by script:eval() function are int, long, float, double, bool and string.